tools/syntax/migx.vim

```diff
@@ -0,0 +1,30 @@
+" Vim syntax file
+" Language: MIGraphX Intermediate Representation
+" Current Maintainer: Charlie Lin (https://github.com/CharlieL7)
+" Previous Maintainer:
+" Last Change: 14 Feb 2024
+
+" quit when a syntax file was already loaded
+if exists("b:current_syntax")
+  finish
+endif
+
+syn keyword migx_keyword param
+syn keyword migx_keyword return
+syn keyword migx_keyword target_id
+syn match migx_ins_number "@\d\+"
+syn match migx_instruction "\s=\s\zs.\+\ze\[" nextgroup=migx_attributes
+syn match migx_attributes "\(\[.\{-}\]\)"
+syn match migx_output_type "\s->\s\zs.\{-}\ze," nextgroup=migx_output_dims
+syn match migx_output_dims "\s\zs{.\{-}}\ze,\starget_id"
+syn match migx_exec_time ":\s\zs.\{-}ms\ze"
+
+let b:current_syntax = "migx"
+
+hi def link migx_keyword NonText
+hi def link migx_ins_number Number
+hi def link migx_instruction Operator
+hi def link migx_attributes Comment
```

Comment:
migx_attributes isn't always present, from what I see: example, the mul operator:
```
@13514 = reshape[dims={-1}](@13513) -> int64_type, {1}, {1}, target_id=0
@13515 = mul(@11820,@11819) -> int64_type, {1}, {1}, target_id=0
```